addons/account/models/account_partial_reconcile.py

```diff
@@ -137,6 +141,14 @@ def create(self, vals_list):
         partials = super().create(vals_list)
         self._pay_matched_payment(partials)
         self._update_matching_number(partials.debit_move_id + partials.credit_move_id)
+        for partial in partials:
+            invoice, payment = (
+                (partial.debit_move_id.move_id, partial.credit_move_id.move_id.origin_payment_id)
+                if partial.credit_move_id.move_id.origin_payment_id
+                else (partial.credit_move_id.move_id, partial.debit_move_id.move_id.origin_payment_id)
+            )
+            if payment not in invoice.matched_payment_ids:
```

Comment:
i suppose that payment has always a value, but just in case maybe we can add:
if payment and payment not in invoice.matched_payment_ids. 
I had in mind that using the in operator on record set was not working and we needed to use the id with the invoice.matched_payment_ids.ids that great but maybe it has changed :thinking: